### ISSUE TITLE
Wrong base url leads to error in production

### ISSUE THREAD

**Issue by tmlmt** [harlan-zw/nuxt-og-image#26]
I've faced quite a few bugs when implementing nuxt-og-image in dev but haven't been able to create a proper repro and have found my way around things. However I am facing a breaking error in production as I cannot access the `/__og_image__/og.png` as I can in dev.

I can see the following error in my pm2 logs:
```
[nuxt] [request error] [unhandled] [500] request to http://www.mywebsite.com:3000/api/og-image-options?path=/en failed, reason: connect ETIMEDOUT 11.22.33.44:3000 (http://www.mywebsite.com:3000/api/og-image-options?path=/en)
  at $fetchRaw2 (/var/www/myproject/.output/server/node_modules/ofetch/dist/shared/ofetch.502a4799.mjs:180:24)  
  at /var/www/myproject/node_modules/nuxt-og-image/dist/runtime/nitro/middleware/og.png.mjs:16:19  
  at Object.handler (/var/www/myproject/.output/server/node_modules/h3/dist/index.mjs:1247:19)  
  at Server.toNodeHandle (/var/www/myproject/.output/server/node_modules/h3/dist/index.mjs:1322:7)
```
I've correctly set up (I think!) the siteUrl via runtimeConfig with https and no port number, so I'm surprised to see that the url which causes an error is in http and with port 3000. I wonder if the following part of the code has something to do with it?

![image](https://user-images.githubusercontent.com/10244927/226533257-8bad430c-233c-441b-8af9-2e129534ed9c.png)

Using Nuxt 3.3.1 and a freshly installed nuxt-og-image.

**Comment by harlan-zw** [comment]
Hey @tmlmt 

You can try out the v2 beta to see if it works any better `2.0.0-beta.0`. 

It's not fully tested so you may run into some other issues.

**Comment by tmlmt** [comment]
Hey @harlan-zw, thanks for tackling this one ! `2.0.0-beta.1` is not working. 
Get the following error: 
```
[Vue warn]: Unhandled error during execution of setup function                                                                                                                                                                                             22:59:41
  at <[[slug]] onVnodeUnmounted=fn<onVnodeUnmounted> ref=Ref< undefined > >
```
Seems to be coming from https://github.com/harlan-zw/nuxt-og-image/blob/e08d1e7f1b8ae9a4874ca0a7ec7fc7490ed2bde2/src/runtime/composables/defineOgImage.ts#L29 that you've added. Are you maybe missing an import of useRuntimeConfig, like in 8444dce?

**Comment by tmlmt** [comment]
Looking forward to your new beta release :))

**Comment by harlan-zw** [comment]
Please try 2.0.0-beta.4 :)

**Comment by tmlmt** [comment]
Almost there? Now I strangely get a Nuxt 500 error when I try to access the og.png image of my homepage:

![image](https://user-images.githubusercontent.com/10244927/227726993-eeddf9d4-481a-4c06-abe9-3278411f6ccd.jpeg)

**Comment by harlan-zw** [comment]
> Almost there? Now I strangely get a Nuxt 500 error when I try to access the og.png image of my homepage:

That's a new one! Any reproduction available :pray: ?

**Comment by tmlmt** [comment]
Hey @harlan-zw, I found out that this was due to one of the `<nuxt-img>` tags in my custom template. I am having some issues with using images: as this is unrelated to this issue (which you seems to have solved with `2.0.0-beta.7` -- thank you for that :+1:), I will close it and open a new one.

**Comment by Sandros94** [comment]
@tmlmt I was following the issue since I had the same problem (but still didn't have time to test the betas).

Until you create the new issue (I don't understand if you will open it here or on the nuxt-img repo) could you explain what problem did you have with `<nuxt-img>`? I'm using both of them too, so you got me curious.
